doc/source/administrator/authentication.md

```diff
@@ -0,0 +1,427 @@
+information before you can configure the helm chart for authentication.
+
+#### GitHub
+
+GitHub is the largest hosting service for git repositories. It is free
+to create an account at GitHub, and relatively straightforward to set up
+OAuth credentials so that users can authenticate with their GitHub
+username/password.
+
+To create OAuth credentials on GitHub, follow these steps:
+
+-   Click your profile picture -> settings -> developer settings
+-   Make sure you're on the "OAuth Apps" tab, then click "New OAuth
+    App"
+-   Fill out the forms (you'll need your hub address) and generate your
+    ID/Secret.
+
+To enable GitHub authentication, add the following to your
+`config.yml`:
+
+```yaml
+hub:
+  config:
+    OAuthenticator:
+      client_id: your-client-id
+      client_secret: your-client-secret
+      oauth_callback_url: https://your-jupyterhub-domain/hub/oauth_callback
+    JupyterHub:
+      authenticator_class: oauthenticator.github.GitHubOAuthenticator
+```
+
+Make sure that the `oauth_callback_url` matches the one
+you set in GitHub.
+
+##### Giving access to organizations on GitHub
+
+You can also restrict access to all of the members of one or more GitHub
+organizations. To do so, see the configuration below.
+
+```yaml
+hub:
+  config:
+    GitHubOAuthenticator:
+      allowed_organizations:
+        - my-github-organization
+    OAuthenticator:
+      scope:
+        - read:user
+```
+
+`scope` can take other values as described in the [GitHub OAuth scopes
+documentation](https://developer.github.com/apps/building-oauth-apps/understanding-scopes-for-oauth-apps/)
+but we recommend `read:user` as this requires no additional
+configuration by GitHub organisations and users. For example, omitting
+the scope means members of an organisation must [set their membership to
+Public](https://docs.github.com/en/github/setting-up-and-managing-your-github-user-account/publicizing-or-hiding-organization-membership)
+to login, whereas setting it to `read:org` may require approval of the
+application by a GitHub organisation admin. Please see [this
+issue](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/issues/687)
+for further information.
+
+```{note}
+Changing `scope` will not change the scope for existing OAuth tokens,
+you must invalidate them.
+```
+
+#### Google
+
+Google authentication is used by many universities (it is part of the
+"G Suite"). Note that using Google authentication requires your Hub to
+have a domain name (it cannot **only** be accessible via an IP address).
+
+```yaml
+hub:
+  config:
+    OAuthenticator:
+      client_id: your-client-id.apps.googleusercontent.com
+      client_secret: your-client-secret
+      oauth_callback_url: https://your-jupyterhub-domain/hub/oauth_callback
+    GoogleOAuthenticator:
+      hosted_domain:
+        - your-university.edu
+      login_service: Your university
+    JupyterHub:
+      authenticator_class: oauthenticator.GoogleOAuthenticator
+```
+
+#### CILogon
+
```

Comment:
Could you add a couple of sentences on what CILogon is, since it's the only probvider that doesn't have any background.

Reply:
```suggestion
Please see CyberInfrastructure Logon's [website](https://www.cilogon.org) for more information about what kind of identity is managed by CILogon.

```